lib/sentry/client_error.ex

```diff
@@ -17,15 +17,22 @@ defmodule Sentry.ClientError do
   @doc """
   The exception struct for a Sentry error.
   """
-  defexception [:reason]
+  defexception [:reason, :http_response]
 
   @typedoc """
   The type for a Sentry error exception.
   """
-  @type t :: %__MODULE__{reason: reason()}
+  @type t :: %__MODULE__{
+          reason: reason(),
+          http_response:
+            nil | {status :: 100..199, headers :: [{String.t(), String.t()}], body :: binary()}
+        }
 
   @type reason() ::
           :too_many_retries
+          | :server_error
```

Comment:
This felt convoluted but needed to pass :server_error by itself as a type so we don't return the headers twice when we call `new`

Reply:
Add `server_error/3`, it's gonna make things simpler 🙃 `new/*` are private anyway so we don't need to care too much, we can always change later.

```elixir
def server_error(status, headers, body) do
  %__MODULE__{reason: :server_error, http_response: {status, headers, body}}
end
```